floris/simulation/farm.py

```diff
@@ -81,10 +81,10 @@ class Farm(BaseClass):
 
     turbine_definitions: list = field(init=False, validator=iter_validator(list, dict))
 
-    turbine_fCts: Dict[str, interp1d] | List[interp1d] = field(init=False, default=[])
-    turbine_fCts_sorted: NDArrayFloat = field(init=False, default=[])
+    turbine_fCts: Dict[str, interp1d] | List[interp1d] = field(init=False, factory=list)
+    turbine_fCts_sorted: NDArrayFloat = field(init=False, factory=list)
```

Comment:
For instances like this below, I would recommend modifying slightly to ensure the typing is valid (even if unchecked currently) to use the following paragdigm:

```python
x: NDArrayFloat = field(init=False, factory=list, converter=floris_array_converter)
```

Essentially what will happen if typing is implemented, without converting to the array type, it will fail. Additionally it creates some confusion (though some already exists tbc) to create a list for an object indicated to be an array.

The only place this will cause issues is with non-float arrays, in which case I'd recommend having two variations of `floris_array_converter` such as `float_array_converter` and `object_array_converter`. I don't know that it needs to have `floris_` at the front at this point, and might be a lingering (but dated) idea from early in the design process.

Because this is a suggested course correction somewhat late in the game, i could see this being a separate issue/PR since that usage is pretty common.